filament/src/main/java/net/fabricmc/filament/task/base/FileOutputTask.java

```diff
@@ -0,0 +1,16 @@
+package net.fabricmc.filament.task.base;
+
+import javax.inject.Inject;
+
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.OutputFile;
+
+public abstract class FileOutputTask extends FilamentTask {
+	@OutputFile
+	public abstract RegularFileProperty getOutputFile();
+
+	@Inject
+	public FileOutputTask() {
+		getOutputFile().finalizeValueOnRead();
```

Comment:
Are these really needed for tasks? I can't imagine anything mutating them *during* task execution

Reply:
Not really, especially for filament. This should really be the default IMO.